packages/material-ui/src/styles/useThemeProps.d.ts

```diff
@@ -16,7 +17,7 @@ export interface AdditionalThemeProps<Theme> {
 }
 
 export default function useThemeProps<
-  Theme extends ThemeWithProps,
```

Comment:
What exactly is failing? Can't you do:

```ts
import { Theme } from '@material-ui/core/styles';

useThemeProps<Theme>({ props: inProps, name: 'MuiPickersToolbarText' })
```

Reply:
`useThemeProps` require 3 arguments, so you need to do this.

```js
useThemeProps<Theme, PickersToolbarTextProps, 'MuiPickersToolbarText'>({
  props: inProps,
  name: 'MuiPickersToolbarText'
})
```

For me, I feels like fixing `ThemeWithProps` with `Theme` is a better solution because that is the implmentation of `useThemeProps` (to return the whole theme)

Reply:
updated to use generic